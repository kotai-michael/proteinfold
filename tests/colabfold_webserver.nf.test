nextflow_pipeline {

    name "Test colabfold webserver mode stub"
    script "../main.nf"
    tag "pipeline"
    tag "test_colabfold_webserver"
    profile "test_colabfold_webserver"

    test("-profile test_colabfold_webserver") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            // Early failure no need to test the rest of snapshots
            assert workflow.success
            assertAll(
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we test pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_proteinfold_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path
                ).match() }
            )
        }
    }
}
