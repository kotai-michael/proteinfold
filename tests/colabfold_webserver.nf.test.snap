{
    "-profile test_colabfold_webserver": {
        "content": [
            5,
            {
                "COLABFOLD_BATCH": {
                    "colabfold_batch": "1.5.2"
                },
                "GENERATE_REPORT": {
                    "python": "3.12.7",
                    "generate_report.py": "Python 3.12.7"
                },
                "Workflow": {
                    "nf-core/proteinfold": "v1.2.0dev"
                }
            },
            [
                "colabfold",
                "colabfold/webserver",
                "colabfold/webserver/T1024_colabfold.pdb",
                "colabfold/webserver/T1024_coverage.png",
                "colabfold/webserver/T1024_mqc.png",
                "colabfold/webserver/T1024_relaxed_rank_01.pdb",
                "colabfold/webserver/T1024_relaxed_rank_02.pdb",
                "colabfold/webserver/T1024_relaxed_rank_03.pdb",
                "colabfold/webserver/T1026_colabfold.pdb",
                "colabfold/webserver/T1026_coverage.png",
                "colabfold/webserver/T1026_mqc.png",
                "colabfold/webserver/T1026_relaxed_rank_01.pdb",
                "colabfold/webserver/T1026_relaxed_rank_02.pdb",
                "colabfold/webserver/T1026_relaxed_rank_03.pdb",
                "colabfold/webserver/top_ranked_structures",
                "generate",
                "generate/test_LDDT.html",
                "generate/test_alphafold2_report.html",
                "generate/test_seq_coverage.png",
                "multiqc",
                "multiqc/multiqc_data",
                "multiqc/multiqc_plots",
                "multiqc/multiqc_report.html",
                "pipeline_info",
                "pipeline_info/nf_core_proteinfold_software_mqc_versions.yml"
            ],
            [
                "T1024_colabfold.pdb:md5,d41d8cd98f00b204e9800998ecf8427e",
                "T1024_coverage.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                "T1024_mqc.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                "T1024_relaxed_rank_01.pdb:md5,d41d8cd98f00b204e9800998ecf8427e",
                "T1024_relaxed_rank_02.pdb:md5,d41d8cd98f00b204e9800998ecf8427e",
                "T1024_relaxed_rank_03.pdb:md5,d41d8cd98f00b204e9800998ecf8427e",
                "T1026_colabfold.pdb:md5,d41d8cd98f00b204e9800998ecf8427e",
                "T1026_coverage.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                "T1026_mqc.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                "T1026_relaxed_rank_01.pdb:md5,d41d8cd98f00b204e9800998ecf8427e",
                "T1026_relaxed_rank_02.pdb:md5,d41d8cd98f00b204e9800998ecf8427e",
                "T1026_relaxed_rank_03.pdb:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test_LDDT.html:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test_alphafold2_report.html:md5,d41d8cd98f00b204e9800998ecf8427e",
                "test_seq_coverage.png:md5,d41d8cd98f00b204e9800998ecf8427e"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-07-24T16:55:02.279523773"
    }
}